pennylane/optimize/momentum_qng_qjit.py

```diff
@@ -0,0 +1,117 @@
+# limitations under the License.
+"""Quantum natural gradient optimizer with momentum for Jax/Catalyst interface"""
+
+from pennylane import math
+
+from .qng_qjit import QNGOptimizerQJIT
+
+
+class MomentumQNGOptimizerQJIT(QNGOptimizerQJIT):
+    r"""Optax-like and ``jax.jit``/``qml.qjit``-compatible implementation of the :class:`~.MomentumQNGOptimizer`,
+    a generalized Quantum Natural Gradient (QNG) optimizer considering a discrete-time Langevin equation
+    with QNG force.
+
+    For more theoretical details, see the :class:`~.MomentumQNGOptimizer` documentation.
+
+    .. note::
+
+        Please be aware of the following:
+
+            - As with ``MomentumQNGOptimizer``, ``MomentumQNGOptimizerQJIT`` supports a single QNode to encode the objective function.
+
+            - ``MomentumQNGOptimizerQJIT`` does not support any QNode with multiple arguments. A potential workaround
+              would be to combine all parameters into a single objective function argument.
+
+            - ``MomentumQNGOptimizerQJIT`` does not work correctly if there is any classical processing in the QNode circuit
+              (e.g., ``2 * theta`` as a gate parameter).
+
+    **Example:**
+
+    Consider a hybrid workflow to optimize an objective function defined by a quantum circuit.
+    To make the optimization faster, the entire workflow can be just-in-time compiled using
+    the :func:`~.qjit` decorator:
+
+    .. code-block:: python
+
+        import pennylane as qml
+        import jax.numpy as jnp
+
+        @qml.qjit(autograph=True)
+        def workflow():
+            dev = qml.device("lightning.qubit", wires=2)
+
+            @qml.qnode(dev)
+            def circuit(params):
+                qml.RX(params[0], wires=0)
+                qml.RY(params[1], wires=1)
+                return qml.expval(qml.Z(0) + qml.X(1))
+
+            opt = qml.MomentumQNGOptimizerQJIT(stepsize=0.1, momentum=0.2)
+
+            params = jnp.array([0.1, 0.2])
+            state = opt.init(params)
+            for _ in range(100):
+                params, state = opt.step(circuit, params, state)
+
+            return params
+
+    >>> workflow()
+    Array([ 3.14159265, -1.57079633], dtype=float64)
+
+    Keyword Args:
+        stepsize=0.01 (float): the stepsize hyperparameter
+        momentum=0.9 (float): the momentum coefficient hyperparameter
+        approx="block-diag" (str): approximation method for the metric tensor.
+
+            - If ``None``, the full metric tensor is computed
+
+            - If ``"block-diag"``, the block-diagonal approximation is computed, reducing
+              the number of evaluated circuits significantly
+
+            - If ``"diag"``, the diagonal approximation is computed, slightly
+              reducing the classical overhead but not the quantum resources
+              (compared to ``"block-diag"``)
+
+        lam=0 (float): metric tensor regularization to be applied at each optimization step
```

Comment:
Nice! Apologies I missed this with the `QNGOptimizerQJIT` PR, but can we move the example below the code example? We'd need to fix the QNGOptimizer docstring as well :). Feel free to do that in a follow-up PR or in this one, doesn't matter to me.

Reply:
Hey @isaacdevlugt, thanks for the comment! If I understand correctly what you mean, this should be done both for `QNGOptimizerQJIT` and `MomentumQNGOptimizerQJIT` docstrings in [8d6684e](https://github.com/PennyLaneAI/pennylane/pull/7606/commits/8d6684ecaa0b17ff624da388972ac7b2ef2157e4).